packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -194,4 +233,20 @@ export class FileNavigatorWidget extends FileTreeWidget {
         </div>;
     }
 
+    protected createWorkspaceRoot(workspace: FileStat, children: TreeNode[]): DirNode {
+        const id = workspace.uri;
```

Comment:
it can be just hardcoded, like `id: 'workspace'` and `WorkspaceNode.id` can just check whether `id` is `workspace`.

Reply:
Sure. If a new types of `WorkspaceNode` and `WorkspaceRootNode` are defined, I wouldn't need `WorkspaceService` to expose the fileStat of `roots.json`.

I will change `WorkspaceService` in the way that Sven suggested: only expose `roots` (and `root` before the `preferences` package is update to support multi root)